app/apps/server/bridges/livechat.ts

```diff
@@ -127,6 +127,7 @@ export class AppLivechatBridge extends LivechatBridge {
 			email: '',
 			connectionData: undefined,
 			phone: {},
+			id: visitor.id,
```

Comment:
The **username** needs to be optional too. When it comes to Omnichannel, it doesn't make much sense to request this information, the serverside handles this following the format `@guest-XXX` .

Currently, every App is using a different approach to create usernames and this is not a good UX.

Reply:
This would require a new release on the apps-engine side to make the username property in `IVisitor` object optional, so we can take this up as a separate task in the next release

Reply:
Changing this would break the compatibility of the API, so we'll most definitely want to avoid that kind of change ☹️ But by checking [the source code of the function](https://github.com/RocketChat/Rocket.Chat/blob/apps/change-storage-strategy/app/livechat/server/lib/Livechat.js#L260) it looks like just setting `visitor.username = ''` would be enough to trigger the behavior